plugins/kubernetes-backend/src/cluster-locator/types.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface ClusterDetails {
+  name: string;
+  url: string;
+  // TODO this will eventually be configured by the auth translation work
+  serviceAccountToken: string | undefined;
+}
+
+export interface KubernetesClusterLocator {
+  getClusterByServiceId(serviceId: string): Promise<ClusterDetails[]>;
```

Comment:
Wouldn't that mean we would have to send the whole entity to the backend? So it would probably then need to be a POST

Reply:
Maybe, although it could also be that the k8s backend makes a request to the catalog backend to fetch the entity first, and you just pass kind/namespace/name in the request.

I'm just a bit concerned with what using `serviceId` as the integration mechanism implies. I'm for example assuming that it requires a deeper integration into the k8s resources and labelling of all resources, and you couldn't do a thinner integration like e.g. `kubernetes.io/namespace` annotations in the Backstage entity.

Would also be happy to merge this PR and discuss more in-depth in a dedicated issue though

Reply:
I think it's definitely an idea to add more info and options for identifying which object belong to a service. IMO this is just one option that is easy to do for now and we could easily add more options later.

Reply:
Yep, not suggesting we get all things in place in this PR, just wanna make sure we're on the same page :grin: